src/components/CardList/Item.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import Card from '../Card';
+import { INodeItem } from './CardList';
+
+export interface Props {
+  node: INodeItem,
+  selectedId?: number,
+  onToggle?(): void,
+  onSelect?(): void,
+}
+
+export interface State {
+  isToggled: boolean,
+}
+
+class Item extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isToggled: false,
+    };
+  }
+
+  onSelect = () => {
+    alert('Id =' + this.props.node.id + ' ' + 'Name =' + this.props.node.name);
+  }
+
+  onToggle = () => {
+    const isToggled = !this.state.isToggled;
+    this.setState(() => ({
+      isToggled,
+    }));
+  }
+
+  render() {
+
+    let childnodes = null;
```

Comment:
`childnodes` and `expandOrCollapse` should be in the component state instead of variables. Then they will be updated this the this.setState() method.